driver-core/src/main/com/mongodb/client/model/Accumulators.java

```diff
@@ -321,6 +492,24 @@ private static <TExpression> BsonField accumulatorOperator(final String name, fi
         return new BsonField(fieldName, new SimpleExpression<TExpression>(name, expression));
     }
 
+    private static <InExpression, NExpression> BsonField pickNAccumulator(
+            final String fieldName, final String accumulatorName, final InExpression inExpression, final NExpression nExpression) {
+        return new BsonField(fieldName, new Document(accumulatorName, new Document("input", inExpression).append("n", nExpression)));
```

Comment:
For builders and operations, we typically have preferred `BsonDocument` to  `Document`, since the former is less susceptible to user-controlled changes to the codec registry.

We should use `BsonDocument` here unless there is a good reason not to, even though doing so adds a bit of  implementation complexity (see `Aggregates.SimplePipelineStage` for an example of this).

Reply:
Right, we can and probably should use `BsonDocument` when the types we have are compatible with it. However, that's not the case here: all three new methods `pickNAccumulator`, `sortingPickAccumulator`, `sortingPickNAccumulator` have to put any type (an expression) inside a `BsonField`, so they have to rely on codec registry. `Aggregates.SimplePipelineStage` does the same [here](https://github.com/mongodb/mongo-java-driver/blob/9fcdefd48faaa9c7ab43ee16c8c215ce05453c2e/driver-core/src/main/com/mongodb/client/model/Aggregates.java#L835). The approach I used in this PR is no different from the one I used in search API.

Reply:
I agree that it's the same as in Search API, but I don't think it's the same as in SimplePipelineStage.   In that case the `Bson` instance came from the application.  In this case the driver is creating the `Document`.

Since we're already doing this elsewhere, I won't object further here.